common/network.c

```diff
@@ -417,3 +417,22 @@ network_routing_rules_set_all_main(bool flush)
 				      "lookup", IP_ROUTING_TABLE, NULL };
 	return proc_fork_and_execvp(argv2);
 }
+
+list_t *
+network_get_physical_interfaces_new()
+{
+	struct if_nameindex *if_ni, *i;
+	if_ni = if_nameindex();
+
+	list_t *if_name_list = NULL;
+
+	IF_NULL_RETVAL(if_ni, NULL);
+
+	for (i = if_ni; i->if_index != 0 || i->if_name != NULL; i++) {
```

Comment:
maybe directly && ?
(if_index == 0 && if_name != NULL possible?

Reply:
this code snipped is according to manpage